lection6/task3.js

```diff
@@ -0,0 +1,52 @@
+// 5*. Напишите рекурсивную функцию, которая принимает на вход число и складывает его цифры. 
+//   Если сумма получилась больше 9 - снова сложите цифры. И так пока, сумма не станет меньше либо равной 9. 
+//   После окончания сложений возвращает полученное число. Например при подаче числа 19 (1+9=10>9, потому 1+0=1) выводится 1
+
+function recursion(number) {
+    if (number <= 9) {
+        return number;
+    } else {
+        let sum = 0;
+        const array = number.toString().split('');
+
+        array.forEach(element => {
+            sum = sum + +element;
+        });
+
+        return recursion(sum);
+    }
+  }
+  
+  console.log(recursion(19123));
+
+// 6*. Написать функцию, которая принимает на вход строку с текстом, и заменяет каждую пару стоящих подряд идентичных букв на одну следующую в алфавите, 
+//     и так пока в тексте не останется двух одинаковых букв стоящих рядом (через пробел и другой знак препинания можно)
+//     Пример: aabc => bbc => cc => d
+
+function transformer(string) {  // работает и с верхним регистром
```

Comment:
try this 
console.log(transformer("baac")); 

should return baac => bbc => cc => d